docs/core/plugins.mdx

```diff
@@ -0,0 +1,279 @@
+---
+title: "Plugins"
+description: "Understanding Suga's plugin system - the building blocks of cloud infrastructure"
+---
+
+Plugins are the fundamental building blocks of Suga platforms. Each plugin is a self-contained, reusable unit that knows how to provision and manage a specific type of cloud resource. Platforms compose multiple plugins to create complete deployment targets.
+
+## What is a Plugin?
+
+A plugin is a package containing:
+
+1. **Terraform Module** - Infrastructure-as-code that provisions cloud resources
+2. **Manifest** - Schema defining configuration properties and outputs
+3. **Runtime Adapter** (optional) - Go code that translates abstract SDK operations into cloud-specific API calls
+
+```
+plugin-directory/
+├── manifest.yaml       # Plugin schema and metadata
+├── main.tf             # Terraform resources
+├── variables.tf        # Input variables
+├── outputs.tf          # Output values
+└── runtime/            # Optional runtime adapters
+    ├── client.go       # SDK client implementation
+    └── operations.go   # Cloud-specific operations
+```
+
+## Plugin Types
+
+Suga uses plugins for different types of infrastructure:
+
+### Resource Plugins
+
+Map to application resources defined in projects:
+
+- **Service plugins** - Compute resources (Lambda, Fargate, Cloud Run)
+- **Bucket plugins** - Object storage (S3, Cloud Storage)
+- **Database plugins** - SQL databases (RDS, Cloud SQL, Neon)
+- **Entrypoint plugins** - CDN and routing (CloudFront, Cloud CDN)
+
+### Identity Plugins
+
+Provide authentication and authorization:
+
+- **IAM role plugins** - AWS IAM roles
+- **Service account plugins** - GCP service accounts
+- Auto-generated based on resource access patterns
+
+### Infrastructure Plugins
+
+Provide foundational cloud resources:
+
+- **VPC plugins** - Network configuration
+- **Load balancer plugins** - Traffic distribution
+- **Security group plugins** - Firewall rules
+
+## How Platforms Use Plugins
+
+Platforms reference plugins in their resource blueprints:
+
+```yaml title="Platform blueprint using plugins"
+services:
+  lambda:
+    source:
+      library: suga/aws
+      plugin: lambda
+    properties:
+      memory: ${self.memory}
+      timeout: ${self.timeout}
+    variables:
+      memory:
+        type: number
+        default: 512
+      timeout:
+        type: number
+        default: 10
+    identities:
+      - source:
+          library: suga/aws
+          plugin: iam-role
+```
+
+When you run `suga build`, the platform:
+
+1. Maps your project resources to plugin blueprints
+2. Resolves property values and variables
+3. Generates Terraform module calls
+4. Configures dependencies and outputs
+
+## Official Plugin Libraries
```

Comment:
Keeping this list up to date might become a bit of a pain, could we link to the public plugin browser with the official filter on?

Reply:
There is no plugin browser yet, but I agree we should replace this with that link, I'll create an issue